src/vimeoplayer/services/ApiLoader.js

```diff
@@ -0,0 +1,76 @@
+import $ from 'jquery';
+
+var DEFAULTS = {
+	url: 'https://player.vimeo.com/api/player.js'
+};
+
+class ApiLoader {
+
+	constructor(options) {
+		this.options = $.extend({}, DEFAULTS, options);
+	}
+
+	requestPlayer() {
+		var self = this;
+
+		// If already requested, return deffered object
+		if (self._deferred) {
+			return self._getDeferred();
+		}
+
+		// If player api was already loaded
+		if (self._hasPlayer()) {
+			return self._getPlayer();
+		}
+
+		// Append player api script
+		self._getScript(self.options.url, function() {
+			self._getPlayer();
+		});
+
+		return self._getDeferred();
+	}
+
+	_hasPlayer() {
+		return (window.Vimeo && window.Vimeo.Player);
+	}
+
+	_getPlayer() {
+		return this._getDeferred().resolve(window.Vimeo.Player);
+	}
+
+	_getDeferred() {
+		this._deferred = this._deferred || $.Deferred();
+		return this._deferred;
+	}
+
+	_getScript(url, callback) {
+		// If the script tag allready exists
+		if ($('script[src="' + this.options.url + '"]').length) {
+			return;
+		}
+
+		var script = document.createElement('script');
```

Comment:
Yes, but you can test if the requested namespace `window.Vimeo` is available which solves two cases:
* If the script tag was loaded by jQuery and removed later on
* If the script tag was attached by the developer in the html file (no need to load this async by the loader)